### PR TITLE
Update DpdkDriverImpl.java to support DPDK trunk interfaces

### PR DESCRIPTION
**Pull request by mbrashearnttglobalnet**
Update to support trunk interfaces for DPDK enabled hosts.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java [hunk 1]
```diff
@@ -70,9 +70,15 @@ public void addDpdkPort(String bridgeName, String port, String vlan, DpdkHelper.
                 dpdkPortVhostUserClientType;
 
         StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(String.format("ovs-vsctl add-port %s %s " +
-                "vlan_mode=access tag=%s " +
-                "-- set Interface %s type=%s", bridgeName, port, vlan, port, type));
+        
+        if (Integer.parseInt(vlan) > 0 && Integer.parseInt(vlan) < 4095) {
+            stringBuilder.append(String.format("ovs-vsctl add-port %s %s " +
+                    "vlan_mode=access tag=%s " +
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java:76]: could you put only this vland_mode and tag in the if-block and the rest around it. The other parts of the string do not depend on the condition and there is nothing that should go in the else-block.

Comment (mbrashearnttglobalnet) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java]: updated

**Comment by nvazquez** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java [hunk 2]
```diff
@@ -70,10 +70,12 @@ public void addDpdkPort(String bridgeName, String port, String vlan, DpdkHelper.
...
-        stringBuilder.append(String.format("ovs-vsctl add-port %s %s " +
-                "vlan_mode=access tag=%s " +
-                "-- set Interface %s type=%s", bridgeName, port, vlan, port, type));
-
+        stringBuilder.append(String.format("ovs-vsctl add-port %s %s ", bridgeName, port));
+        if (Integer.parseInt(vlan) > 0 && Integer.parseInt(vlan) < 4095) {
+            stringBuilder.append(String.format("vlan_mode=access tag=%s ", vlan));
+        }
+        stringBuilder.append(String.format("-- set Interface %s type=%s", port, type));
+
```
Comment (nvazquez) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java:78]: Packages failing with: 
````
[ERROR] /jenkins/workspace/acs-pr-centos7-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.14.0.0-SNAPSHOT/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java:78: Line has trailing spaces. [RegexpSingleline]
````

@mbrashearnttglobalnet can you refactor this line space as a linux carriage?

Comment (mbrashearnttglobalnet) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/dpdk/DpdkDriverImpl.java]: removed the white spaces on the blank line.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@nvazquez you ok like this?
@blueorangutan test

**Comment by rhtyd** [comment]
Travis has passed, our smoketests test only kvm with bridge driver.
Merging based on this.

**Pull request merged**